### PR TITLE
feat(docs): migrate llms.txt generation to consume docs map

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Refactors `renderLlmsIndex()` and `renderLlmsFull()` to accept a `DocsMap` instead of raw `ExpectedOutput`
- Pipeline becomes: discover → build expected output → generate docs map → render llms from docs map
- Adds adapter layer to convert docs map entries back to the render format
- Golden snapshot tests ensure llms.txt output is identical before and after the migration

## Test plan

- [ ] `bun run test --filter=@outfitter/docs` passes
- [ ] `bun run docs:check:ci` confirms llms.txt output hasn't changed
- [ ] Snapshot tests validate the rendered format matches golden files
- [ ] `renderLlmsIndex()` and `renderLlmsFull()` signatures use `DocsMap`

Closes OS-279

Closes: OS-279

<h3>Greptile Summary</h3>

Refactored llms.txt generation to consume `DocsMap` instead of internal `ExpectedOutput`, adding an adapter layer that reads files from disk. The new `renderLlmsIndexFromMap()` and `renderLlmsFullFromMap()` functions are exported alongside the existing internal functions, enabling future CLI commands to operate on the docs map directly. Comprehensive tests validate the rendering format matches expected behavior, including proper handling of missing files, path security, and temporary directory cleanup.

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with minimal risk
- Clean refactoring that adds new functions without modifying existing behavior. Comprehensive test coverage validates rendering format, path security, and edge cases. Previous test issue with temp directories was already addressed by developer.
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/docs/package.json | Added export for new `core/docs-map-render` module |
| packages/docs/src/index.ts | Exported `renderLlmsFullFromMap` and `renderLlmsIndexFromMap` from new docs-map-render module |
| packages/docs/src/core/docs-map-render.ts | New module that adapts DocsMap to llms.txt rendering; properly validates paths and handles missing files |
| packages/docs/src/__tests__/docs-map-render.test.ts | Comprehensive test coverage for rendering functions; temp directory cleanup properly implemented |

</details>

</details>

<sub>Last reviewed commit: d238714</sub>

### CODE REVIEW

File: packages/docs/src/core/docs-map-render.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    .filter((e) => e.package)
+    .sort((a, b) => a.outputPath.localeCompare(b.outputPath));
+
+  for (const entry of sortedEntries) {
+    const absolutePath = resolve(workspaceRoot, entry.outputPath);
+    let content: string;
+    try {
+      content = await readFile(absolutePath, "utf8");
+    } catch {
+      continue; // Skip if file doesn't exist yet
```
[the remark this answers is not in the document]
Comment (galligan) [reply packages/docs/src/core/docs-map-render.ts]: Addressed in subsequent commits on this stack; this thread is now outdated relative to the current diff. Resolving.
